File: buerguer-pig/AMarotoChivite-BurguerPig/AMarotoChivite-BurguerPig-Java/src/main/java/Ficheros/BurguerPig/models/Ingredient.java
package Ficheros.BurguerPig.models;

import java.io.Serializable;

public class Ingredient implements Serializable {
    private String name;
    private Double price;
    private Integer id;

    private static int counter = 0;

    public Ingredient(String name, double price) {
        this.name = name;
        this.price = price;
        this.id = autoCount();
    }

    public Ingredient(String name, double price, int id) {
        this.name = name;
        this.price = price;
        this.id = id;
    }

    public Integer getID() {
        return id;
    }

    public String getName() {
        return name;
    }

    public Double getPrice() {
        return price;
    }

    @Override
    public String toString() {
        return "Ingredient(name='" + name + "', price=" + price + ", id=" + id + ")";
    }

    private static int autoCount() {
        return counter++;
    }
}

File: buerguer-pig/AMarotoChivite-BurguerPig/AMarotoChivite-BurguerPig-Kotlin/src/main/kotlin/Ficheros/BurguerPig/models/Ingredient.kt
package Ficheros.BurguerPig.models

import java.io.Serializable

class Ingredient(val name: String, val price: Double) : Serializable {
    private var id: Int = autoCount()

    constructor(name: String, price: Double, id: Int) : this(
        name,
        price
    ) {
        this.id = id
    }

    fun getID(): Int {
        return id
    }

    override fun toString(): String {
        return "Ingredient(name='$name', price=$price, id=$id)"
    }

    companion object {
        var counter = 0
        private fun autoCount(): Int {
            return counter++
        }
    }
}